File: ojpython/interview/virtu/183_hex.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

#include <iostream>
#include <sstream>
using namespace std;

string solution(string &S) {
    // write your code in C++14 (g++ 6.2.0)
    std::stringstream stream;
    stream << std::hex << 257;
    std::string result( stream.str() );
    return result;
}

int main()
{
    string s=string("257");
    cout<<solution(s);
}

File: ojpython/interview/virtu/183_hex.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(S):
    # write your code in Python 3.6
    hexstr = "%X" % int(S)
    encode = {'1':'I','0':'O','A':'A','B':'B','C':'C','D':'D','E':'E','F':'F'}

    lst = list(hexstr)
    #print(lst)
    result = []
    for c in lst:
        if c in encode:
            result.append(encode[c])
        else:
            #print(c)
            return "ERROR"

    return ''.join(result)

if __name__=="__main__":
    print(solution(257)=="IOI")
    print(solution(3)=="ERROR")



